core/infrastructure/configmap.go

```diff
@@ -16,50 +16,190 @@ package infrastructure
+	resources, err := kubernetes.ResourceC(c.Client).ListAll(objectTypes, owner.GetNamespace(), owner)
 	if err != nil {
-		return
+		return nil, err
 	}
-	if !exists {
-		s.recorder.Eventf(s.Client, owner, corev1.EventTypeWarning, "NotExists", "ConfigMap %s does not exist", key.Name)
-		return
+	kubernetesResources := resources[reflect.TypeOf(corev1.ConfigMap{})]
+	configMapList := make([]*corev1.ConfigMap, len(kubernetesResources))
+	for i, kubernetesResource := range kubernetesResources {
+		configMapList[i] = kubernetesResource.(*corev1.ConfigMap)
 	}
-	if framework.IsOwner(cm, owner) {
-		return
+	return configMapList, nil
+}
+
+func (c *configMapHandler) MountConfigMapOnDeployment(deployment *appsv1.Deployment, configMap *corev1.ConfigMap) error {
+	if isMountAsFile(configMap) {
+		if err := mountAsFile(deployment, configMap); err != nil {
+			return err
+		}
 	}
-	if err = framework.AddOwnerReference(owner, s.Scheme, cm); err != nil {
-		return
+
+	if isMountAsLiteral(configMap) {
+		mountAsLiteral(deployment, configMap)
 	}
-	if err = kubernetes.ResourceC(s.Client).Update(cm); err != nil {
+	return nil
+}
+
+func isMountAsFile(configMap *corev1.ConfigMap) bool {
+	return configMap.Annotations[FromFileKey] == "true"
+}
+
+func mountAsFile(deployment *appsv1.Deployment, configMap *corev1.ConfigMap) error {
+	if err := appendVolumeIntoDeployment(deployment, configMap); err != nil {
+		return err
+	}
+	appendVolumeMountIntoDeployment(deployment, configMap)
+	return nil
+}
+
+func appendVolumeIntoDeployment(deployment *appsv1.Deployment, cm *corev1.ConfigMap) (err error) {
+	for _, volume := range deployment.Spec.Template.Spec.Volumes {
+		if volume.Name == cm.Name {
+			return
+		}
+	}
+
+	fileMode, err := strconv.ParseInt(cm.Annotations[FileModeKey], 0, 32)
+	if err != nil {
 		return
 	}
-	return true, nil
+	fileModeInt32 := int32(fileMode)
+
+	// append volume if its not exists
+	deployment.Spec.Template.Spec.Volumes =
+		append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: cm.Name,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					DefaultMode: &fileModeInt32,
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cm.Name,
+					},
+				},
+			},
+		})
+	return nil
+}
+
+func appendVolumeMountIntoDeployment(deployment *appsv1.Deployment, cm *corev1.ConfigMap) {
+	for fileName := range cm.Data {
+		mountPath := path.Join(cm.Annotations[MountPathKey], fileName)
+		for _, volumeMount := range deployment.Spec.Template.Spec.Containers[0].VolumeMounts {
+			if volumeMount.MountPath == mountPath {
+				return
+			}
+		}
+
+		// append volume mount if its not exists
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts =
+			append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+				Name:      cm.Name,
+				MountPath: mountPath,
+				SubPath:   fileName,
+			})
+	}
+}
+
+func isMountAsLiteral(configMap *corev1.ConfigMap) bool {
```

Comment:
shouldn't it actually query the installed config maps ans return the ones with matches the label?
if, not probably the func name needs to be updated.

Reply:
@spolti I am always got confused while naming my function. This function reads configmap labels and check whether it created using literals or through file. 
Could you please suggest some better name.

Reply:
changed name to `isConfigMapCreatedUsingLiteral`